src/UIState/MainMenu.cpp

```diff
@@ -236,37 +236,40 @@ void MainMenu::selectSet() {
 // pH=7.325 B 7.125
 // T=12.25 H 12.75
 void MainMenu::idle() {
-  PHControl *phControl = PHControl::instance();
-  char output[20];
-  output[0] = 'p';
-  output[1] = 'H';
-  output[2] = millis() / 1000 % 2 ? '=' : ' ';
-  dtostrf(PHProbe::instance()->getPh(), 5, 3, output + 3);
-  output[8] = ' ';
-  output[9] = phControl->isOn() ? 'B' : ' ';
-  output[10] = ' ';
-  dtostrf(PHControl::instance()->getTargetPh(), 5, 3, output + 11);
-  LiquidCrystal_TC::instance()->writeLine(output, 0);
-  TemperatureControl *tempControl = TemperatureControl::instance();
-  TempProbe_TC *tempProbe = TempProbe_TC::instance();
-  float temp = tempProbe->getRunningAverage();
-  if (temp < 0.0) {
-    temp = 0.0;
-  } else if (99.99 < temp) {
-    temp = 99.99;
-  }
-  char status = tempControl->isHeater() ? 'h' : 'c';
-  if (tempControl->isOn()) {
-    status = toupper(status);
+  if (millis() - lastDisplayTime > 1000) {
```

Comment:
I had expected that instead of only updating the display every second, we would modify `LiquidCrystal_TC::writeLine()` so that if the new text was the same as the old text we would skip the write. I guess I'm curious to know if the calculation of the output is expensive or the final `print()` statement in `LiquidCrystal_TC.cpp`.

Reply:
Currently the LCD `print()` is called twice per second in MainMenu, so it doesn't seem worse than calling it twice in a second when the text changes. Regardless of whether the bottleneck is in the LCD `print()` or the calculation, it seems to follow that we wouldn't want to calculate things more than we need to; in this case I think it's appropriate to say, "If we're not going to display more than one screen per second, we might as well only calculate one per second."

Reply:
First, the idle menu might update more often than once per second, e.g., turning the bubbler on and off (though this is a minor matter). Second, when we leave the idle menu we go back to updating the display every time through the loop, so we are back to adding unnecessary delays. 

If making this change seems out of scope, then just create an issue and reference it from this one. I'd be fine with that.